src/test/java/de/muenchen/mobidam/MobilithekRouteS3Test.java

```diff
@@ -83,17 +88,14 @@ class MobilithekRouteS3Test {
     @MockBean
     private SstManagementIntegrationService sstService;
 
-    @MockBean
-    private EnvironmentReader envReader;
-
     @MockBean
     private MimeTypeProcessor mimeTypeProcessor;
 
     @Captor
     private ArgumentCaptor<DatentransferCreateDTO> datentransferCaptor;
 
-    @EndpointInject("mock:s3Destination")
-    private MockEndpoint mockS3Endpoint;
+    @MockBean
```

Comment:
Wenn der EnvironmentReader wieder eine Bean ist, dann kannst Du den mocken und musst nicht mehr den Provider mocken. Dadurch wird der Test auf die Komplettroute ein Stück vollständiger.

Reply:
Was den EnvironmentReader angeht, der im Lib-Modul des Spring-Boot Starter Comons steckt, war ich etwas unsicher, wie mit dem umzugehen ist. Da hätte ich wohl besser gefragt. Jedenfalls habe ich mich bemüht das Lib-Modul mit seinen Abhängigkeiten so einfach und ohne Spring Abhängigkeiten zu halten. 
Der EnvironmentReader wird nur im S3CredentialProvider des Lib-Moduls verwendet. 
Ich füge dann jetzt wieder alle Spring Abhängigkeiten im Lib-Modul hinzu, um aus dem EnvironmentReader per @Component eine Bean zu machen (import org.springframework.core.env.Environment; import org.springframework.stereotype.Component;) , so wie es ursprünglich mal war, als ich es aus der S3 EAI in den Starter verschoben habe ?  Dann kann ich den EnvironmentReader im Lib-Modul als Bean dem S3CredentialProvider (ebenfalls als Bean per @Component initialisiert) mit geben. 
Habe ich das jetzt richtig verstanden habe?

Reply:
Nein, die Lib kannst Du so lassen, aber die Lese-Methode sollte nicht mehr static sein und im starter initialisierst du eine Bean EnvironmentReader.

Reply:
In mobidam-eai-commons-starter:0.0.2 ist der EnvironmentReader wieder eine Bean.